lib/solidus_subscriptions/processor.rb

```diff
@@ -98,6 +98,12 @@ def new_installments(user)
           sub.advance_actionable_date
           sub.cancel! if sub.pending_cancellation?
           sub.deactivate! if sub.can_be_deactivated?
+          if SolidusSubscriptions.configuration.clear_past_installments
+            sub.installments.unfulfilled.each do |installment|
+              installment.actionable_date = nil
+              installment.save!
+            end
+          end
           sub.installments.create!
```

Comment:
These 4 lines should be part of a transaction, I think: if creating the new installment fails, then the previous updates should be rolled back.

Reply:
Isn't this already covered by the transaction just above? https://github.com/solidusio-contrib/solidus_subscriptions/blob/ca929b4c77766a1c2a5eadbaa659583fbb51ca45/lib/solidus_subscriptions/processor.rb#L95

Reply:
It is, damn GH diffs!